src/main/java/com/salesforce/dataloader/controller/Controller.java

```diff
@@ -281,91 +335,84 @@ private void copyFile(File file, File destFile) {
+        }
 
-            // TODO: we should log creating new files/directories
-            // TODO: why did we add this?
-            File configDir = new File(configDirPath);
-            if (!(configDir.isDirectory() || configDir.mkdirs())) {
-                throw new ControllerInitializationException("could not create configuration directory: " + configDir);
-            } else {
-                logger.info("config dir created at " + configDir.getAbsolutePath());
+        // Create dir if it doesn't exist
+        boolean isMkdirSuccessful = createDir(configDir);
+        if (!isMkdirSuccessful) {
+            throw new ControllerInitializationException(Messages.getMessage(getClass(), "errorCreatingOutputDir", configDir));
+        }
+        appPath = configDir.getAbsolutePath();
+
+        // check if the config file exists
+        File configFile = new File(appPath, CONFIG_FILE);
+        String configPath = configFile.getAbsolutePath();
+        if (!configFile.exists()) {
+
+            File defaultConfigFile = new File(getDefaultConfigDir(), DEFAULT_CONFIG_FILE);
+
+            // If default config exists, copy the default to user config
+            // If doesn't exist, create a blank user config in "./target"
+            if (defaultConfigFile.exists()) {
+                try {
+                    // Copy default config to user config
+                    logger.info(String.format("User config file does not exist in '%s' Default config file is copied from '%s'",
+                            configPath, defaultConfigFile.getAbsolutePath()));
+                    Files.copy(defaultConfigFile.toPath(), configFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+                } catch (IOException e) {
+                    String errorMsg = String.format("Failed to copy '%s' to '%s'", defaultConfigFile.getAbsolutePath(), configFile);
+                    logger.warn(errorMsg, e);
+                    throw new ControllerInitializationException(errorMsg, e);
+                }
             }
-
-            // let the File class combine the dir and file names
-            File configFile = new File(configDirPath, CONFIG_FILE);
-
-            // TODO: why did we add this?
-            if (!configFile.exists()) try {
-                configFile.createNewFile();
-                configFile.setWritable(true);
-                configFile.setReadable(true);
-                logger.info("config file created at " + configFile.getAbsolutePath());
-            } catch (IOException e) {
-                throw new ControllerInitializationException(e);
+            else {
+                try {
+                    // Create a blank user config in "./target"
+                    appPath = new File(ProgramDirectoryUtil.getProgramDirectory(), "target").getAbsolutePath();
+                    configFile = new File(appPath, CONFIG_FILE);
```

Comment:
If we get here, then a "proper" config dir exists (possibly having just been created, but still) under the user profile. Given that, why not use it? Why create a new "target" subdirectory of either the install location (which may just fail with access denied) or the working directory (which seems to be what you intend to do, but isn't always what `getProgramDirectory` returns), when you've got a config directory ready to go?